### DISCUSSION TITLE
Forecast tomorrow 487kWh!!!

### DISCUSSION THREAD

**Discussion by C00K50N** [nathanmarlor/foxess_em#104] [Q&A]
Not sure what's going on here, but my forecast is miles off.  

I now have the latest version v1.4.8 after having to restore my HA onto a new SD card (tried to update HA core, realised I didn't have enough space on my old card, bit of faffing but now restored onto a bigger card and updated to latest HA and Core versions).

Prior to the restore, I had noticed my forecasts were in the 100s of kWh but figured it was because I'd had trouble updating to v1.4.8 and had rolled back to a previous version.  I presumed when I was up and running on the bigger SD card and then uninstalled and reinstalled FoxESS-EM it would sort itself out.  But not so, it seems.  Screenshot below of all sensors.  Any idea what I've done wrong @nathanmarlor?
![image](https://user-images.githubusercontent.com/102669305/204766112-b6b832e0-b71b-42e0-90d8-8a3f521dd00e.png)

NB My forecasts on the Solcast site itself look about right:
![image](https://user-images.githubusercontent.com/102669305/204766319-4c9bad95-373e-440c-a302-1b2ec7de1665.png)

**Comment by nathanmarlor** [comment]
How odd. The code to grab the forecasts is probably the simplest in the whole integration. They look to be a factor of 100 off? 

Could you grab the data manually through the Solcast toolkit (get the data button) for estimated actuals and forecast and paste here please?

**Reply by C00K50N** [reply]
Is this what you mean? 

{"estimated_actuals":[{"pv_estimate":0.1736,"period_end":"2022-11-30T10:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3332,"period_end":"2022-11-30T10:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2748,"period_end":"2022-11-30T09:30:00.0000000Z","period":"PT30M"},{"pv_estimate":1.5828,"period_end":"2022-11-30T09:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0822,"period_end":"2022-11-30T08:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T08:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T07:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T07:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T06:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T06:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T05:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T05:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T04:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T04:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T03:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T03:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T02:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T02:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T01:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T01:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T00:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-30T00:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T23:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T23:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T22:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T22:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T21:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T21:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T20:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T20:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T19:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T19:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T18:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T18:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T17:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T17:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T16:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0158,"period_end":"2022-11-29T16:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2101,"period_end":"2022-11-29T15:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.7001,"period_end":"2022-11-29T15:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.4959,"period_end":"2022-11-29T14:30:00.0000000Z","period":"PT30M"},{"pv_estimate":1.1396,"period_end":"2022-11-29T14:00:00.0000000Z","period":"PT30M"},{"pv_estimate":1.4476,"period_end":"2022-11-29T13:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.9187,"period_end":"2022-11-29T13:00:00.0000000Z","period":"PT30M"},{"pv_estimate":1.6089,"period_end":"2022-11-29T12:30:00.0000000Z","period":"PT30M"},{"pv_estimate":2.8403,"period_end":"2022-11-29T12:00:00.0000000Z","period":"PT30M"},{"pv_estimate":3.2562,"period_end":"2022-11-29T11:30:00.0000000Z","period":"PT30M"},{"pv_estimate":3.0438,"period_end":"2022-11-29T11:00:00.0000000Z","period":"PT30M"},{"pv_estimate":2.8892,"period_end":"2022-11-29T10:30:00.0000000Z","period":"PT30M"},{"pv_estimate":2.5945,"period_end":"2022-11-29T10:00:00.0000000Z","period":"PT30M"},{"pv_estimate":2.4261,"period_end":"2022-11-29T09:30:00.0000000Z","period":"PT30M"},{"pv_estimate":1.6334,"period_end":"2022-11-29T09:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.241,"period_end":"2022-11-29T08:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T08:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T07:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T07:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T06:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T06:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T05:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T05:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T04:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T04:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T03:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T03:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T02:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T02:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T01:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T01:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T00:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-29T00:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T23:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T23:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T22:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T22:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T21:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T21:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T20:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T20:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T19:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T19:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T18:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T18:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T17:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T17:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"period_end":"2022-11-28T16:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0227,"period_end":"2022-11-28T16:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.1983,"period_end":"2022-11-28T15:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.7104,"period_end":"2022-11-28T15:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.8537,"period_end":"2022-11-28T14:30:00.0000000Z","period":"PT30M"},{"pv_estimate":1.5411,"period_end":"2022-11-28T14:00:00.0000000Z","period":"PT30M"},{"pv_estimate":2.2718,"period_end":"2022-11-28T13:30:00.0000000Z","period":"PT30M"},

and

{"forecasts":[{"pv_estimate":0.2012,"pv_estimate10":0.1578,"pv_estimate90":0.3782,"period_end":"2022-11-30T11:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3589,"pv_estimate10":0.1775,"pv_estimate90":0.646,"period_end":"2022-11-30T11:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.4423,"pv_estimate10":0.1972,"pv_estimate90":0.7345,"period_end":"2022-11-30T12:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3974,"pv_estimate10":0.2051,"pv_estimate90":0.7156,"period_end":"2022-11-30T12:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2919,"pv_estimate10":0.1561,"pv_estimate90":0.6358,"period_end":"2022-11-30T13:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2469,"pv_estimate10":0.1132,"pv_estimate90":0.5662,"period_end":"2022-11-30T13:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2341,"pv_estimate10":0.0898,"pv_estimate90":0.7218,"period_end":"2022-11-30T14:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.1873,"pv_estimate10":0.0663,"pv_estimate90":0.5785,"period_end":"2022-11-30T14:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.1288,"pv_estimate10":0.0468,"pv_estimate90":0.3316,"period_end":"2022-11-30T15:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0585,"pv_estimate10":0.0234,"pv_estimate90":0.1188,"period_end":"2022-11-30T15:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0078,"pv_estimate10":0.0039,"pv_estimate90":0.0117,"period_end":"2022-11-30T16:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T16:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T17:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T17:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T18:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T18:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T19:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T19:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T20:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T20:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T21:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T21:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T22:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T22:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T23:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-11-30T23:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T00:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T00:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T01:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T01:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T02:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T02:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T03:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T03:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T04:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T04:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T05:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T05:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T06:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T06:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T07:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T07:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T08:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0161,"pv_estimate10":0.004,"pv_estimate90":0.1259,"period_end":"2022-12-01T08:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0887,"pv_estimate10":0.0161,"pv_estimate90":1.5076,"period_end":"2022-12-01T09:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.1714,"pv_estimate10":0.0319,"pv_estimate90":2.2073,"period_end":"2022-12-01T09:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2816,"pv_estimate10":0.0513,"pv_estimate90":2.6047,"period_end":"2022-12-01T10:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3974,"pv_estimate10":0.0671,"pv_estimate90":2.9464,"period_end":"2022-12-01T10:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.471,"pv_estimate10":0.0742,"pv_estimate90":3.163,"period_end":"2022-12-01T11:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.5282,"pv_estimate10":0.082,"pv_estimate90":3.1788,"period_end":"2022-12-01T11:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.5564,"pv_estimate10":0.0811,"pv_estimate90":3.0213,"period_end":"2022-12-01T12:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.5816,"pv_estimate10":0.0888,"pv_estimate90":2.8365,"period_end":"2022-12-01T12:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.6005,"pv_estimate10":0.0966,"pv_estimate90":2.5698,"period_end":"2022-12-01T13:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.5627,"pv_estimate10":0.0966,"pv_estimate90":2.1897,"period_end":"2022-12-01T13:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.4678,"pv_estimate10":0.0888,"pv_estimate90":1.7719,"period_end":"2022-12-01T14:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3469,"pv_estimate10":0.0734,"pv_estimate90":1.2606,"period_end":"2022-12-01T14:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2029,"pv_estimate10":0.0507,"pv_estimate90":0.738,"period_end":"2022-12-01T15:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0976,"pv_estimate10":0.0234,"pv_estimate90":0.1962,"period_end":"2022-12-01T15:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0118,"pv_estimate10":0.0039,"pv_estimate90":0.0162,"period_end":"2022-12-01T16:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T16:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T17:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T17:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T18:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T18:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T19:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T19:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T20:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T20:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T21:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T21:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T22:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T22:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T23:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-01T23:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T00:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T00:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T01:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T01:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T02:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T02:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T03:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T03:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T04:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T04:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T05:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T05:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T06:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T06:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T07:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T07:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0,"pv_estimate10":0,"pv_estimate90":0,"period_end":"2022-12-02T08:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0121,"pv_estimate10":0,"pv_estimate90":0.122,"period_end":"2022-12-02T08:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.0725,"pv_estimate10":0.0121,"pv_estimate90":1.4099,"period_end":"2022-12-02T09:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.141,"pv_estimate10":0.0242,"pv_estimate90":2.1684,"period_end":"2022-12-02T09:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.2113,"pv_estimate10":0.0359,"pv_estimate90":2.7671,"period_end":"2022-12-02T10:00:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3036,"pv_estimate10":0.0478,"pv_estimate90":3.0562,"period_end":"2022-12-02T10:30:00.0000000Z","period":"PT30M"},{"pv_estimate":0.3846,"pv_estimate10":0.0552,"pv_estimate90":3.0647,"period_end":"2022-12-02T11:00:00.0000000Z","period":"PT30M"}]}

**Reply by nathanmarlor** [reply]
Yep, that's it. I was hoping to see some extreme values in there but everything checks out. 

Random one - is your datetime/timezone set right in HA?

**Reply by C00K50N** [reply]
Looks to be correct, yes.  Location is set correctly, and timezone is GMT.
![image](https://user-images.githubusercontent.com/102669305/204789031-01a4d137-4625-40f3-a4ba-eeb4659a8d7d.png)

Should I change these to specify format?  Would that have a bearing?
![image](https://user-images.githubusercontent.com/102669305/204789239-d9306190-29bd-4605-a8d5-ae32b9a39f70.png)

**Reply by nathanmarlor** [reply]
Very strange! The only thing I can suggest is deleting the integration, deleting the forecast entities and restarting. It's almost as if it's getting the units of measurement wrong... but none of that has changed since the very first release?!

**Reply by nathanmarlor** [reply]
Another suggestion - upgrade to the latest beta which includes graphing then add the apex chart to see what's happening to the forecast data...

**Reply by nathanmarlor** [reply]
One final suggestion... Add the forecast to the energy dashboard. This is just a dictionary of the raw values received from Solcast.

**Reply by C00K50N** [reply]
Hmm...  yeah forecast is showing the ridiculously high values too:
![image](https://user-images.githubusercontent.com/102669305/204843455-6bde803e-e8e4-44e9-868d-6bdf7b46f487.png)

I'll try deleting all and restarting again.  Feeling increasingly out of my depth here, not long after I had it almost set up the way I wanted it.

**Reply by C00K50N** [reply]
How do I go about deleting the forecast entities?  The only forecast one listed here isn't related and can't be deleted it seems...
![image](https://user-images.githubusercontent.com/102669305/204848069-11affc3d-b1af-49a7-aab0-cfaac25ba033.png)

**Reply by nathanmarlor** [reply]
I'm not 100% sure to be honest, not something I've needed to tackle yet. I think HA might keep a reference of entities removed through an integration but I'm not sure for how long etc. Can you remember when it started happening? Any particular event or triggers?

**Reply by nathanmarlor** [reply]
https://github.com/nathanmarlor/foxess_em/blob/main/custom_components/foxess_em/forecast/forecast_model.py#L51-L56

FYI this is the forecast logic. I honestly can't see any reason it would be so out. You've got one single Solcast site setup right?

**Reply by C00K50N** [reply]
Yep, just the one Solcast site.  Unless you include the demo sites on Solcast, but they didn't appear to cause a problem before.  I've just deleted them in case it makes a difference.

I first noticed the forecast was wrong when I tried to update to v1.4.8 but then couldn't properly install it.  If you recall on another thread I found it didn't show up on the Integrations page to allow me to configure it.  So I removed it again via HACS and was able to reinstall a previous version.  You suggested I check the HA Core version and upgrade, but I couldn't straight away because I didn't have the space on my 8GB SD card.  So I ordered a larger card, then backed up and restored onto the new card (and seem to have lost all my history in the process, but that's a different story).

I have deleted what I can, restarted HA and also rebooted my rpi.  Going to try reinstalling the integration now and see what happens...

**Reply by C00K50N** [reply]
Hmm...  different issue after the reinstall a couple hours ago.  I don't know if it's because I tried the beta version?
![image](https://user-images.githubusercontent.com/102669305/204891384-185e5afb-fb02-42da-a246-60afddb820a1.png)

**Reply by nathanmarlor** [reply]
It does load, forecast then battery. Looks like load has worked as you have legit looking load sensors. Have you exceeded the Solcast limit? Should be in the logs.

**Reply by C00K50N** [reply]
Ah, yes. Looks like you’re right, I’ve used 50 already. I’ll check it again in the morning and see what it’s doing.

**Reply by C00K50N** [reply]
That's better!! 
![image](https://user-images.githubusercontent.com/102669305/204999886-f929b737-ecfe-4f4f-bf6d-8d01e6671a4e.png)

I wonder if it had been including the demo Solcast sites as well as my own site.  I hadn't worked out how to delete the demo ones previously, and it hadn't been a problem.  But maybe I did something different in re-adding the API on one of the re-installs.

Either way, I'm happy that it now looks like it's showing correctly.

**Reply by nathanmarlor** [reply]
Amazing! I had a hunch that'd be the problem as 1.4.x moved to using the Solcast API key to grab all available sites.

Glad you're up and running again. Got a few more tweaks to make to the beta before I do a full release but it's not far off. Definitely worth checking out the new graph functionality!

![Screenshot_20221201-081903.png](https://user-images.githubusercontent.com/3436911/205001516-a75f4614-d931-4625-944d-cbffd79b0175.png)

House load in pink, solar in orange, battery in blue and grid in green :)

**Reply by nathanmarlor** [reply]
Btw your house load has jumped up considerably - have you forgotten to exclude an Eddi/Zappi?

**Reply by C00K50N** [reply]
I do love a graph!  Will take a look at that later today... I'll need to install ApexCharts card will I?  I'll work it out later on.

Our house load is always high, especially in winter as our heating and hot water is all electric.  I'm sure I excluded our Zappi on setup so I think it'll be right.  Looking back we used c50kWh yesterday without a car charge.

**Reply by C00K50N** [reply]
How do I go about implementing the graph functionality @nathanmarlor?  Are you able to share the code you used to build that card above as a starting point?

**Reply by nathanmarlor** [reply]
https://github.com/nathanmarlor/foxess_em/blob/main/apex-example/apex.yaml

:)

Will do a proper readme before release.
